src/Common/src/System/Net/Http/Http2/Hpack/HPackEncoder.cs

```diff
@@ -151,7 +152,7 @@ private bool EncodeString(string value, Span<byte> destination, out int bytesWri
                         for (int i = 0; i < value.Length; i++)
                         {
                             char c = value[i];
-                            destination[i] = (byte)((uint)(c - 'A') <= ('Z' - 'A') ? c | 0x20 : c);
+                            destination[i] = (byte)(lowercase && (uint)(c - 'A') <= ('Z' - 'A') ? c | toLowerMask : c);
```

Comment:
@Tratcher isn't this lowercasing headers? Didn't you mention this is invalid?

Reply:
This is the fix for the bug that was lower casing headers. When called for header names it should lower case, but for header values it should not.

Reply:
oops 😄

Reply:
It might lead to better code if the `&&` is hoisted outside the loop:

```cs
int lowerOrKeepCaseMask = lowercase ? 0x20 : 0x00;
for (...)
{
    char c = value[i];
    destination[i] = (byte)((uint)(c - 'A') <= ('Z' - 'A') ? c | lowerOrKeepCaseMask : c);
    ...
}
```